generate.go

```diff
@@ -646,8 +648,53 @@ func getDefaultTemplate() specs.Spec {
 			},
 		},
 		Hostname: "shell",
-		Mounts:   []specs.Mount{},
+		Mounts: []specs.Mount{
+			{
+				Destination: "/proc",
+				Type:        "proc",
+				Source:      "proc",
+				Options:     nil,
+			},
+			{
+				Destination: "/dev",
+				Type:        "tmpfs",
+				Source:      "tmpfs",
+				Options:     []string{"nosuid", "strictatime", "mode=755", "size=65536k"},
+			},
+			{
+				Destination: "/dev/pts",
+				Type:        "devpts",
+				Source:      "devpts",
+				Options:     []string{"nosuid", "noexec", "newinstance", "ptmxmode=0666", "mode=0620", "gid=5"},
```

Comment:
The `gid=5` option will be [problematic for user namespaces][1] (where you can only map a single user).  From [user_namespaces(7)][2] for unprivileged (in the parent user namespace) processes:

> The data written to uid_map (gid_map) must consist of a single line that maps the writing process's effective user ID  (group ID)  in  the  parent user namespace to a user ID (group ID) in the user namespace.

So if the caller maps themselves to 0, then all other GIDs (including 5) will be unmapped.  Are we not compliance testing unprivileged processes until runC sorts out its required filesystems support?  If we are, how should I handle this mount entry?

[1]: https://github.com/opencontainers/runc/issues/225#issuecomment-136519577
[2]: http://man7.org/linux/man-pages/man7/user_namespaces.7.html